Tribler/Core/APIImplementation/LaunchManyCore.py

```diff
@@ -761,10 +761,18 @@ def network_shutdown(self):
 
     def save_download_pstate(self, infohash, pstate):
         """ Called by network thread """
+
+        # if pstate can't be found
+        if not pstate.get('state', 'engineresumedata'):
```

Comment:
Yes, but you only call `save_resume_data` if `'engineresumedata'` is empty so you will keep using the same data. (If libtorrent doesn't generate resume data alerts by its own initiative, but even if this is the case the proposal still stands)